File: Desktop/hotel_system_desktop/lib/models/inventory_item.dart
class InventoryItem {
  final int id;
  final String name;
  final String unit;
  final String category;
  final int inventoryItemCategoryId;
  final int minimumStockLevel;

  InventoryItem({
    required this.id,
    required this.name,
    required this.unit,
    required this.category,
    required this.inventoryItemCategoryId,
    required this.minimumStockLevel,
  });

  factory InventoryItem.fromJson(Map<String, dynamic> json) => InventoryItem(
        id: json['id'] ?? 0,
        name: json['name'] ?? '',
        unit: json['unit'] ?? '',
        category: json['category'] ?? '',
        inventoryItemCategoryId: json['inventoryItemCategoryId'] ?? 0,
        minimumStockLevel: json['minimumStockLevel'] ?? 0,
      );
}

File: backend/eBooking/Persistence/Models/InventoryItem.cs
namespace Persistence.Models
{
    /// <summary>
    /// Referentni artikal skladišta (npr. sapun, peškiri, mini bar napici).
    /// InventoryTransaction se veže na ovu tabelu preko InventoryItemId (FK),
    /// umjesto da je to "goli" broj bez referencijalnog integriteta.
    /// </summary>
    public class InventoryItem : BaseEntity
    {
        public string Name { get; set; } = string.Empty;
        public string Unit { get; set; } = string.Empty;
        public int InventoryItemCategoryId { get; set; }
        public InventoryItemCategory? InventoryItemCategory { get; set; }
        public int MinimumStockLevel { get; set; }
        public ICollection<InventoryTransaction> Transactions { get; set; } = new List<InventoryTransaction>();
    }
}
